File: api/src/main/java/fr/huiitre/tools/modules/riot/valorant/application/skin/view/ValorantSkinView.java
package fr.huiitre.tools.modules.riot.valorant.application.skin.view;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record ValorantSkinView(
    Long id,
    UUID assetId,
    String name,
    String iconUrl,
    UUID tierUuid,
    UUID contentTierUuid,
    Long weaponId,
    List<ValorantSkinLevelView> levels,
    List<ValorantSkinChromaView> chromas,
    boolean owned,
    boolean watched,
    LocalDateTime ownedAt,
    LocalDateTime watchedAt
) {}

File: api/Modules/Riot/Valorant/Application/Skin/Views/ValorantSkinView.cs
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Views;

namespace Tools.Api.Modules.Riot.Valorant.Application.Skin.Views;

// Owned/Watched et leurs dates viennent de jointures sur le compte Valorant courant : les dates
// sont nulles quand le drapeau est faux.
public sealed record ValorantSkinView(
    long Id,
    Guid AssetId,
    string Name,
    string? IconUrl,
    Guid? TierUuid,
    ValorantContentTierView? ContentTier,
    long? WeaponId,
    List<ValorantSkinLevelView> Levels,
    List<ValorantSkinChromaView> Chromas,
    bool Owned,
    bool Watched,
    DateTime? OwnedAt,
    DateTime? WatchedAt
);
